### DISCUSSION TITLE
[Question] Async-sometimes example?

### DISCUSSION THREAD

**Discussion by nemyung** [pmndrs/jotai#1339] [Q&A] [answered]
Hi there!

I was reading `Async sometimes` section in your [docs](https://jotai.org/docs/guides/async#async-sometimes) and making an example on [sandbox](https://codesandbox.io/s/async-sometimes-cfvesc). While everything was perfect, I got a type error. It seems like `Promise<Value>`is not assignable to `SetStateAction<Value>` . Is it a way to solve this problem?

And i found some typo.
- with Jotai are is => with Jotai is
- On the example, Jōtai switches `syncAtom` to `asyncAtom` using `setStateAction` with Promise. But on the docs, it is switching `from async to sync`. I think the sync and async might be swapped.

Thanks!

**Comment by dai-shi** [comment, marked as answer]
You need to annotate the atom type.
Here's the fix: https://codesandbox.io/s/async-sometimes-forked-p1i1f5

Please feel free to open a PR to fix typo. Thanks!

**Reply by nemyung** [reply]
Thanks!
